R/plotNorm.R

```diff
@@ -76,25 +81,137 @@ plotNorm <- function(DGEdata,
-            geom_boxplot(alpha = 0.5) +
-            facet_grid(~Normalization) +
-            ggtitle(stringr::str_c("Log2CPM before/after", normalize, "normalization", sep = " "))  +
-            theme_gray() +
-            theme(axis.text.x = element_blank(),
-                  legend.position = "none")
+    return(resultPlot)
+}
+
+build_cx_data <- function(data) {
+    cx.data <- data.frame()
+    for (sample in levels(as.factor(data$SampleID))) {
+        if (nrow(cx.data) == 0) {
+            cx.data <- select_sample(data, sample, TRUE)
+        } else {
+            cx.data <-
+                cbind(cx.data, select_sample(data, sample))
+        }
+
     }
+    return(cx.data)
+}
+
+select_sample <- function(data, sampleID, select_all = FALSE) {
+    if (select_all) {
+        sample <-
+            subset(data, SampleID == sampleID,
+                   select = c("GeneID", "Normalization", "Log2CPM"))
+        colnames(sample)[3] = sampleID
+    } else {
+        sample <-
+            subset(data, SampleID == sampleID, select = c("Log2CPM"))
+        colnames(sample)[1] = sampleID
+    }
+
+    return(sample)
+}
+
+build_cx_density_plot <- function(data, title) {
+    cx.data <- build_cx_data(data)
+    xlab <- "Log2CPM"
+    ylab <- "density"
+    events <- JS(
+        "{'mousemove' :
+            function(o, e, t) {
+                if (o != null && o != false) {
+                    if (o.objectType == 'Density' &&
+                        o.display != null &&
+                        o.display.indexOf('-') > -1) {
+                        sampleID = o.display.split('-')[2];
+                        t.showInfoSpan(e, '<b>Sample ID</b>: ' + sampleID);
+                    } else {
+                        t.showInfoSpan(e, o.display);
+                    };
+                };
+           }
+        }"
+    )
+    resultPlot <- canvasXpress(
+        data                    = subset(cx.data, select = -c(GeneID, Normalization)),
+        varAnnot                = cx.data[, c("GeneID", "Normalization")],
+        histogramData           = TRUE,
+        graphType               = "Scatter2D",
+        xAxisTitle              = xlab,
+        yAxisTitle              = ylab,
+        hideHistogram           = TRUE,
+        showHistogramDensity    = TRUE,
+        segregateVariablesBy    = list("Normalization"),
+        title                   = title,
+        showLegend              = FALSE,
+        colorScheme             = "ColorScpectrum",
+        events                  = events
+    )
+    return(resultPlot)
+}
+
+build_cx_box_plot <- function(data, title) {
+    cx.data <- build_cx_data(data)
+    xlab <- "Log2CPM"
+    ylab <- "SampleID"
+    y <- as.data.frame(t(subset(cx.data, select = -c(GeneID, Normalization))))
+    resultPlot <- canvasXpress(
+        data                    = y,
+        smpAnnot                = cx.data[, c("GeneID", "Normalization")],
```

Comment:
Refactor variable names more informative. Either rename cx.data e.g smp.data or smp.annot and Y to cx.data.